src/software/ai/hl/stp/play/free_kick/free_kick_play_fsm.cpp

```diff
@@ -0,0 +1,324 @@
+#include "free_kick_play_fsm.h"
+
+FreeKickPlayFSM::FreeKickPlayFSM(const TbotsProto::AiConfig &ai_config)
+    : ai_config(ai_config),
+      align_to_ball_tactic(std::make_shared<MoveTactic>()),
+      shoot_tactic(std::make_shared<KickTactic>()),
+      chip_tactic(std::make_shared<ChipTactic>()),
+      passer_tactic(std::make_shared<KickTactic>()),
+      receiver_tactic(
+          std::make_shared<ReceiverTactic>(ai_config.receiver_tactic_config())),
+      receiver_positioning_tactics(
+          {std::make_shared<MoveTactic>(), std::make_shared<MoveTactic>()}),
+      defense_play(std::make_shared<DefensePlay>(ai_config)),
+      receiver_position_generator(ReceiverPositionGenerator<EighteenZoneId>(
+          std::make_shared<const EighteenZonePitchDivision>(
+              Field::createSSLDivisionBField()),
+          ai_config.passing_config())),
+      pass_generator(ai_config.passing_config()),
+      best_pass_and_score_so_far(
+          PassWithRating{.pass = Pass(Point(), Point(), 0), .rating = 0})
+{
+}
+
+void FreeKickPlayFSM::setupPosition(const Update &event)
+{
+    PriorityTacticVector tactics_to_run = {{}};
+
+    // Passing robot
+    updateAlignToBallTactic(event.common.world_ptr);
+    tactics_to_run[0].emplace_back(align_to_ball_tactic);
+
+    if (event.common.num_tactics <= 1)
+    {
+        LOG(WARNING)
+            << "Not enough tactics to setup pass receivers and defenders during free kick";
+        return;
+    }
```

Comment:
The `align_to_ball_tactic` will not get set if we have `num_tactics == 1`

Reply:
Yeah you're right. I decided to just remove the check. `setReceiverAndDefenderTactics` already can handle the case where `num_tactics = 1`. Though If `num_tactics = 1`, we just chip it towards the center and run towards it since there are no defenders. I hope we won't have to test that in real life :sweat_smile: